src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ProjectPackagesPrintUtility.cs

```diff
@@ -82,193 +82,158 @@ internal static class ProjectPackagesPrintUtility
-                                outdated));
+                valueSelectors.Add(p => new FormattedCell(p.OriginalRequestedVersion));
             }
-            else
+
+            // "Resolved" version
+            valueSelectors.Add(p => new FormattedCell(GetPackageVersion(p)));
+
+            switch (listPackageArgs.ReportType)
             {
-                tableToPrint = await packages.ToStringTableAsync(
-                       headers,
-                       p => Task.FromResult((object)""),
-                       p => Task.FromResult((object)p.Name),
-                       p =>
-                       {
-                           if (p.AutoReference)
-                           {
-                               autoReferenceFound = true;
-                               return Task.FromResult((object)"(A)");
-                           }
-                           return Task.FromResult((object)"");
-                       },
-                       p => Task.FromResult((object)p.OriginalRequestedVersion),
-                       async p => PrintVersion(
-                                p.ResolvedPackageMetadata.Identity.Version,
-                                await p.ResolvedPackageMetadata.GetDeprecationMetadataAsync() != null,
-                                outdated));
+                case ReportType.Outdated:
+                    // "Latest" version
+                    valueSelectors.Add(p => new FormattedCell(GetPackageVersion(p, useLatest: true)));
+                    break;
+                case ReportType.Deprecated:
+                    valueSelectors.Add(p => new FormattedCell(
+                        PrintDeprecationReasons(p.ResolvedPackageMetadata.GetDeprecationMetadataAsync().Result)));
+                    valueSelectors.Add(p => new FormattedCell(
+                        PrintAlternativePackage((p.ResolvedPackageMetadata.GetDeprecationMetadataAsync().Result)?.AlternatePackage)));
+                    break;
+                case ReportType.Vulnerable:
+                    valueSelectors.Add(p => PrintVulnerabilitiesSeverities(p.ResolvedPackageMetadata.Vulnerabilities));
+                    valueSelectors.Add(p => PrintVulnerabilitiesAdvisoryUrls(p.ResolvedPackageMetadata.Vulnerabilities));
+                    break;
             }
 
-            //Handle printing with colors
-            foreach (var text in tableToPrint)
+
+            var tableToPrint = packages.ToStringTable(headers, valueSelectors.ToArray());
+
+            tableHasAutoReference = autoReferenceFlagged;
+            return tableToPrint;
+        }
+
+        internal static void PrintPackagesTable(IEnumerable<FormattedCell> tableToPrint)
+        {
+            foreach (var formattedCell in tableToPrint)
             {
-                Console.Write(text);
+                if (formattedCell.ForegroundColor.HasValue)
+                {
+                    Console.ForegroundColor = formattedCell.ForegroundColor.Value;
```

Comment:
I noticed other parts of the code set the console colours without regard, but it makes me wonder, if someone does `dotnet list package > result.txt`, will the text file have the colour control characters, making it just about unreadable in most text editors?  I also wonder what it looks like if a customer runs it on CI and if AzDO and GitHub build logs colour the output or not.  What does it look like doing `dotnet list package --vulnerable | grep high`.

Anyway, I'm sure in the spirit of doing minial changes that we won't be interested in making these experiences good if they're not alrady.

Reply:
I just tested `> result.txt` and there are no control chars, so it looks like we're good there. I'm on a Windows box so tested `| Select-String -Pattern High` in PS and all was well there too.